### ISSUE TITLE
Consider marking `IncludeLoaderError::cause` as `Send + Sync + 'static`

### ISSUE THREAD

**Issue by LeoniePhiline** [jolimail/mrml-core#45]
Hi!

Again thanks for the library, I still very much enjoy using it!

The issue I am decribing occurs when wrapping an `MrmlParseError` and trying to use it within a `tokio::spawn` (or `thread::spawn`).

It goes back to https://github.com/jolimail/mrml-core/commit/57cd747775f50d1e6a999210710d8dfa0fc871f5 (3 weeks ago)

# **TLDR:**

The issue occurs at 

https://github.com/jolimail/mrml-core/blob/main/src/prelude/parse/loader.rs#L13-L19

where extending `Option<Box<dyn std::error::Error>>` to `Option<Box<dyn std::error::Error + Send + Sync + 'static>>` should make errors wrapping `IncludeLoaderError` usable in multithreaded contexts.

As far as I observed, you should not run into issues with that, as [`std::io::Error`](https://doc.rust-lang.org/std/io/struct.Error.html) as used in the local loader is `Send` and `Sync`, as well as [`ureq::Error`](https://docs.rs/ureq/latest/ureq/enum.Error.html) and [`reqwest::Error`](https://docs.rs/reqwest/latest/reqwest/struct.Error.html) used in the http loader both are `Send` and `Sync`.

Some type annotations (`Option<&dyn std::error::Error>` and `Box<(dyn std::error::Error + 'static)>`) would need to be adjusted in `with_cause()` and `cause()`.

# **Context:**

Here my example error (simplified):

```rust
use mrml::prelude::{parse::Error as MrmlParseError, render::Error as MrmlRenderError};

use crate::mailer::Mailer;

#[derive(thiserror::Error, Debug)]
pub enum MailError<M: Mailer> {
    #[error(transparent)]
    BuilderError(#[from] MailBuilderError),
    #[error(transparent)]
    MailerError(M::Error),
}

#[derive(thiserror::Error, Debug)]
pub enum MailBuilderError {
    #[error("failed to parse MJML template")]
    ParseMrml(#[from] MrmlParseError),
    #[error("failed to render MJML template")]
    RenderMrml(#[from] MrmlRenderError),
}
```

Passing `MailError`, or its inner `MailBuilderError` through a `tokio::spawn` task / thread (if multi-threaded) boundary, causes the following compile error, indicating that the `Send` marker is missing on `mrml::prelude::parse::loader::IncludeLoaderError`:

```
error[E0277]: `(dyn std::error::Error + 'static)` cannot be sent between threads safely
   --> mail/src/mailer/smtp_mailer.rs:193:26
    |
193 |               tokio::spawn(async move {
    |  _____________------------_^
    | |             |
    | |             required by a bound introduced by this call
194 | |                 smtp_mailer
195 | |                     .send_sync(message)
196 | |                     .await
...   |
205 | |                 Ok::<(), MailError<Arc<SmtpMailer>>>(())
206 | |             });
    | |_____________^ `(dyn std::error::Error + 'static)` cannot be sent between threads safely
    |
    = help: the trait `Send` is not implemented for `(dyn std::error::Error + 'static)`
    = note: required for `Unique<(dyn std::error::Error + 'static)>` to implement `Send`
    = note: required because it appears within the type `Box<(dyn std::error::Error + 'static)>`
    = note: required because it appears within the type `Option<Box<(dyn std::error::Error + 'static)>>`
    = note: required because it appears within the type `mrml::prelude::parse::loader::IncludeLoaderError`
    = note: required because it appears within the type `mrml::prelude::parse::Error`
note: required because it appears within the type `error::MailBuilderError`
   --> mail/src/error.rs:14:10
    |
14  | pub enum MailBuilderError {
    |          ^^^^^^^^^^^^^^^^
note: required because it appears within the type `error::MailError<std::sync::Arc<smtp_mailer::SmtpMailer>>`
   --> mail/src/error.rs:6:10
    |
6   | pub enum MailError<M: Mailer> {
    |          ^^^^^^^^^
    = note: required because it appears within the type `Result<(), error::MailError<std::sync::Arc<smtp_mailer::SmtpMailer>>>`
note: required by a bound in `tokio::spawn`
   --> /.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/task/spawn.rs:164:20
    |
164 |         T::Output: Send + 'static,
    |                    ^^^^ required by this bound in `tokio::spawn`
```

Here in color:

![image](https://user-images.githubusercontent.com/22329650/221948637-25df2087-c4f3-4293-adae-827f446937e6.png)

**Comment by jdrouet** [comment]
Thanks for opening this issue!
I'll take a look tomorrow 😉

Did you take a look at https://github.com/jdrouet/catapulte/ ?

**Comment by LeoniePhiline** [comment]
Yes, I'm aware of catapulte. It just isn't the right tool for my use case:

Using a microservice adds unnecessary complexity and indirection for my pure-rust architecture, where integrating a shared mailer lib crate into binaries/executables is not an issue.

**Comment by LeoniePhiline** [comment]
Confirmed fixed by #52 ❤️
